lib/parser/index.js

```diff
@@ -520,6 +520,13 @@ Parser.prototype._insertElement = function (token, namespaceURI) {
     this.openElements.push(element);
 };
 
+Parser.prototype._insertFakeElement = function (tagName) {
+    var element = this.treeAdapter.createElement(tagName, NS.HTML, []);
+
+    this._attachElementToTree(element);
```

Comment:
Wondering if this should be made conditional (depending on some option) to address cases where people don't want fake elements inserted.